RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,213 @@
+---
+title: "Gastric cancer: gene-level summarization with `tximeta`"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+**CCDL 2021**
+
+In this notebook, we'll import the transcript expression quantification output from `salmon quant` using the [`tximeta`](https://bioconductor.org/packages/release/bioc/html/tximeta.html) package.
+`tximeta` is in part a wrapper around another package, [`tximport`](https://bioconductor.org/packages/release/bioc/html/tximport.html), which imports transcript expression data and summarizes it to the gene level.
+Working at the gene rather than transcript level has a number of potential advantages for interpretability, efficiency, and reduction of false positives ([Soneson _et al._ 2016](https://doi.org/10.12688/f1000research.7563.2)).
+`tximeta` eases some of the burden of import by automatically identifying the correct set of annotation data to append to many data sets ([Love _et al._ 2020](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007664)).
+
+![](diagrams/rna-seq_5.png)
+
+For more information about `tximeta`, see [this excellent vignette](https://www.bioconductor.org/packages/devel/bioc/vignettes/tximeta/inst/doc/tximeta.html) from Love _et al_.
+
+## Libraries and functions
+
+```{r library}
+# Load magrittr for the pipe
+library(magrittr)
+
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+```{r directories, live = TRUE}
+# directory where the data are located
+data_dir <- file.path("data", "gastric-cancer")
+
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path(data_dir, "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+We'll need the `quant.sf` files for all the samples in an experiment which we have stored in `quant_dir`. 
```

Comment:
We should explain here what these files are, including not only that this is the output from `salmon`, but also a bit of a mention of the file structure. We might want to also make a note about what `list.files()` is doing here, as I don't think we have covered it before. 

This is also a place where we should start to address https://github.com/AlexsLemonade/training-modules/issues/360, so I am happy to leave this for another PR.

Reply:
Yeah, seems better suited for next PR.